dtos/requests/device.go

```diff
@@ -122,6 +122,9 @@ func ReplaceDeviceModelFieldsWithDTO(device *models.Device, patch dtos.UpdateDev
 	if patch.AutoEvents != nil {
 		device.AutoEvents = dtos.ToAutoEventModels(patch.AutoEvents)
 	}
+	if patch.Protocol != nil {
+		device.Protocol = *patch.Protocol
```

Comment:
we don't allow this to be updated, so ignore this field here.
just leave a comment to indicate we ignore it on purpose.